docs/core/base/document.md

```diff
@@ -0,0 +1,112 @@
+# Document
+
+The term "Document" is used in the only in the code. It refers a single piece of content that created by Elementor, such as a post, page, or a saved template.
+
+Each document has its own type and features.
+
+A document is saved as a `post` in the WordPress `wp_posts` table. His elements and settings are saved as metadata in the `wp_postmeta` table.
+
+In PHP, the document is represented by the `Elementor\Base\Document` class.
+
+This class is like a DB Model that knows how to save and load its data.
+
+### Why "Document"?
+
+WordPress' content are called `posts` it's an historical term that came from the days of WordPress as a blogging system.
+
+In Elementor, it can be many document types for one `post_type` like the saved templates Page and Section that are both a `elementor_library` post type.
+
+The term `document` is used instead, as the browser called his DOM content as a "document".
+
+### Main Features
+
+* Get the data from DB
+* Save the data in DB
+* Get the elements and settings
+* Get metadata like URLs to create, edit and preview the document
+* Get the WP_Post object
+* Get the list of widgets categories to display in the panel
+
+### How to create a new Document type?
+
+A new type of document can be used in order to create special functionality for it.
+
+For example, an `Header` in Elementor Pro is a new type of document that knows to appear in the header of the site.
+
+A new document type should be created by extending the `Elementor\Base\Document` class, and define his type and the post type it will be used in.
```

Comment:
```suggestion
A new document type should be created by extending the `Elementor\Base\Document` class, and defining its type and the post type it will be used in.
```